presto-orc/src/test/java/com/facebook/presto/orc/BenchmarkOrcDecimalReader.java

```diff
@@ -0,0 +1,143 @@
+import com.facebook.presto.orc.metadata.OrcMetadataReader;
+import com.facebook.presto.spi.block.Block;
+import com.facebook.presto.spi.type.DecimalType;
+import com.facebook.presto.spi.type.SqlDecimal;
+import com.google.common.collect.ImmutableMap;
+import io.airlift.units.DataSize;
+import org.joda.time.DateTimeZone;
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.BenchmarkMode;
+import org.openjdk.jmh.annotations.Fork;
+import org.openjdk.jmh.annotations.Measurement;
+import org.openjdk.jmh.annotations.Mode;
+import org.openjdk.jmh.annotations.OutputTimeUnit;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.annotations.TearDown;
+import org.openjdk.jmh.annotations.Warmup;
+import org.openjdk.jmh.runner.Runner;
+import org.openjdk.jmh.runner.options.Options;
+import org.openjdk.jmh.runner.options.OptionsBuilder;
+import org.openjdk.jmh.runner.options.VerboseMode;
+
+import java.io.File;
+import java.io.IOException;
+import java.math.BigInteger;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import static com.facebook.presto.orc.OrcTester.Compression.NONE;
+import static com.facebook.presto.orc.OrcTester.Format.ORC_12;
+import static com.facebook.presto.orc.OrcTester.writeOrcColumn;
+import static com.facebook.presto.spi.type.DecimalType.createDecimalType;
+import static com.google.common.io.Files.createTempDir;
+import static io.airlift.testing.FileUtils.deleteRecursively;
+import static io.airlift.units.DataSize.Unit.MEGABYTE;
+import static java.util.UUID.randomUUID;
+import static org.apache.hadoop.hive.serde2.objectinspector.primitive.PrimitiveObjectInspectorFactory.javaHiveDecimalObjectInspector;
+
+@SuppressWarnings("MethodMayBeStatic")
+@State(Scope.Thread)
+@OutputTimeUnit(TimeUnit.SECONDS)
+@Fork(3)
+@Warmup(iterations = 20, time = 500, timeUnit = TimeUnit.MILLISECONDS)
+@Measurement(iterations = 20, time = 500, timeUnit = TimeUnit.MILLISECONDS)
+@BenchmarkMode(Mode.AverageTime)
+public class BenchmarkOrcDecimalReader
+{
+    public static final DecimalType DECIMAL_TYPE = createDecimalType(30, 10);
+
+    @Benchmark
+    public Object readDecimal(BenchmarkData data)
+            throws Throwable
+    {
+        OrcRecordReader recordReader = data.createRecordReader();
+        List<Block> blocks = new ArrayList<>();
+        while (recordReader.nextBatch() > 0) {
+            Block block = recordReader.readBlock(DECIMAL_TYPE, 0);
+            blocks.add(block);
+        }
+        return blocks;
+    }
+
+    @SuppressWarnings("FieldMayBeFinal")
+    @State(Scope.Thread)
+    public static class BenchmarkData
+    {
+        private File temporary;
+        private File dataPath;
+
+        @Setup
+        public void setup()
+                throws Exception
+        {
+            temporary = createTempDir();
```

Comment:
You can remove the temp file just like with the temp dir. But, I get the point about not being able to write if the file exists (I thought it might just overwrite it).